manifests/plugin/column_view.pp

```diff
@@ -0,0 +1,17 @@
+# = Foreman Column View
+#
+# This class installs the column view plugin and optionally manages the configuration file
+#
+# === Parameters:
+#
+# $columns:: an hash of columns to add to the configuration
+#
+class foreman::plugin::column_view (
+  Optional[Foreman::Column_view] $columns = undef
```

Comment:
The current code would handle ```undef``` and ```{}``` in the same way. This means you always manage the file now. Is that intended? If that's the case then I'd prefer to default to ```{}``` and remove ```Optional``` but otherwise you can store the config in a variable:
```puppet
if $columns {
  $config = template('foreman/foreman_column_view.yaml.erb')
} else {
  $config = undef
}
```

Would you mind adding a trailing comma on this line as well?